lib/mix/lib/mix/dep/fetcher.ex

```diff
@@ -92,22 +89,18 @@ defmodule Mix.Dep.Fetcher do
     # dependency is missing, it could directly affect one of the
     # dependencies we are trying to compile, causing the whole thing
     # to fail.
-    #
-    # If there is any other dependency that is not ok, we include
-    # it for compilation too, this is our best to try to solve the
-    # maximum we can at each deps.get and deps.update.
-    deps =
+    parent_deps =
       if Enum.all?(all_deps, &available?/1) do
         Enum.uniq_by(with_depending(deps, all_deps), & &1.app)
       else
-        deps
+        []
```

Comment:
We don't have to mark these as fetched since they were marked immediately after fetching.